apps/ejabberd/src/mongoose_acc.erl

```diff
@@ -120,3 +126,13 @@ dump(_, []) ->
 dump(Acc, [K|Tail]) ->
     ?ERROR_MSG("~p = ~p", [K, maps:get(K, Acc)]),
     dump(Acc, Tail).
+
+read_child([]) ->
+    {undefined, undefined};
+read_child([Chld]) ->
+    #xmlel{name = Name, attrs = Attrs} = Chld,
+    {xml:get_attr_s(<<"xmlns">>, Attrs), Name};
+read_child(_) ->
+    % children are sometimes longer then 1 but only when it is an error, not sure yet how
```

Comment:
How about `message` stanza with `body` and other sub elements indicating for example the chat state (active, typing etc).